includes/admin/tools/export/class-batch-export-donors.php

```diff
@@ -162,67 +189,114 @@ private function get_cols( $columns ) {
 	 * @return array $data The data for the CSV file.
 	 */
 	public function get_data() {
-
-		$data = array();
-
 		$i = 0;
 
-		if ( ! empty( $this->form ) ) {
+		$data             = array();
+		$cached_donor_ids = Give_Cache::get( $this->query_id, true );
 
-			// Export donors of a specific product.
-			global $give_logs;
+		if ( ! empty( $this->form ) ) {
 
+			// Export donors for a specific donation form and also within specified timeframe
 			$args = array(
-				'post_parent'    => absint( $this->form ),
-				'log_type'       => 'sale',
-				'posts_per_page' => 30,
-				'paged'          => $this->step,
+				'output'     => 'payments', // Use 'posts' to get standard post objects
```

Comment:
@Devinwalker we are getting donor for specific form from payments instead of log